internal/operands/common-templates/reconcile.go

```diff
@@ -87,6 +95,32 @@ func (c *commonTemplates) Cleanup(request *common.Request) error {
 	return nil
 }
 
+func pauseCRs(request *common.Request) {
+	patch := []byte(`{"metadata":{"annotations":{"kubevirt.io/operator.paused": "true"}}}`)
+	var kubevirtCommonTemplatesBundles kvsspv1.KubevirtCommonTemplatesBundleList
+	err := request.Client.List(context.TODO(), &kubevirtCommonTemplatesBundles, &client.ListOptions{})
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error listing common template bundles: %s", err))
+		return
+	}
+	if err == nil && len(kubevirtCommonTemplatesBundles.Items) > 0 {
+		for _, kubevirtCommonTemplatesBundle := range kubevirtCommonTemplatesBundles.Items {
+			err = request.Client.Patch(context.TODO(), &kvsspv1.KubevirtCommonTemplatesBundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: kubevirtCommonTemplatesBundle.ObjectMeta.Namespace,
+					Name:      kubevirtCommonTemplatesBundle.ObjectMeta.Name,
+				},
+			}, client.RawPatch(types.MergePatchType, patch))
+			if err != nil {
```

Comment:
A `NotFound` error is an accepted case here and does not required any error logging, because if the old CR does not exist (during future upgrades), we just proceed with the deployment

Reply:
This will also return an error if the old CRDs are not installed. I'm not sure if it is the same `NotFound` or different.

In a new cluster deployment, there will not be the old operator and old CRDs, so it would be a useless to call this function. I think in case the CRD does not exist, the `request.Client` cannot watch the resource so it cannot do caching. On every reconcile iteration this function would need to call k8s API.

Can you add code to not call this function if old CRDs are not installed? Maybe the check can be done when the operator starts.